contracts/governance/extensions/GovernorVotes.sol

```diff
@@ -4,28 +4,51 @@
 pragma solidity ^0.8.0;
 
 import "../Governor.sol";
-import "../utils/IVotes.sol";
+import "../../interfaces/IERC5805.sol";
 
 /**
  * @dev Extension of {Governor} for voting weight extraction from an {ERC20Votes} token, or since v4.5 an {ERC721Votes} token.
  *
  * _Available since v4.3._
  */
 abstract contract GovernorVotes is Governor {
-    IVotes public immutable token;
+    IERC5805 public immutable token;
 
     constructor(IVotes tokenAddress) {
-        token = tokenAddress;
+        token = IERC5805(address(tokenAddress));
+    }
+
+    /**
+     * See EIP 5805.
+     */
+    function clock() public view virtual override returns (uint48) {
+        try token.clock() returns (uint48 timepoint) {
```

Comment:
Aren't these prone to return bombing?

```solidity
contract Token {
    function clock(bytes4 interfaceId) external pure returns (bytes memory) {
        assembly{
            revert(0, 1_000_000)
        }
    }
}
```

Or is it safe because it only copies `uint48` from the `RETURNDATA`?

Reply:
The Token is not an arbitrary contract. Its one contract that the governor heavily relies on. IMO, the governor should assume the token is safe.

That being said, we could write it in assembly to prevent that. I hate that we often have to do it, because it reduces readability. Not sure if he want/need to do it here.

Reply:
> That being said, we could write it in assembly to prevent that.

Thanks for the explanation, but I don't think it should be done at the moment. However, this is something to consider in 6372's security considerations, isn't it?

Reply:
> However, this is something to consider in 6372's security considerations, isn't it?

EIP-6372 is about what contract should return (to be compliant) ... not what nasty contract might do wrong. As an example, EIP-20 doesn't (and shouldn't) mention return bombing by the `decimals` or `totalSupply` function.

Reply:
I'm solving, but I'll still keep pushing for that security consideration in the EIP. EIP-20 didn't have it because there's no way to backwards update an EIP. Following EIP guidelines:

> Security Considerations: ... E.g. include security-relevant design decisions, concerns, important discussions, implementation-specific guidance and pitfalls, an outline of threats and risks and how they are being addressed. EIP submissions missing the “Security Considerations” section will be rejected. An EIP cannot proceed to status “Final” without a Security Considerations discussion deemed sufficient by the reviewers.